### PR TITLE
Stop playing after this song.

### PR DESCRIPTION
**Pull request by vsarunov**
Added functionality to set a song after which the music should stop playing.

### CODE REVIEW

File: BreadPlayer.Views.UWP/ViewModels/ShellViewModel.cs [hunk 1]
```diff
@@ -608,6 +618,17 @@ void PlayFile(Mediafile toPlayFile, bool play = false)
                     if (play == true)
                         Player.IgnoreErrors = true;
 
+                    if(_songToStopAfter!=null && _songToStopAfter==PreviousSong)
+                    {
+                       await Player.Pause();
```
Comment (theweavrs) [inline BreadPlayer.Views.UWP/ViewModels/ShellViewModel.cs]: Why not call `PlayPause` method directly?

File: BreadPlayer.Views.UWP/ViewModels/ShellViewModel.cs [hunk 2]
```diff
@@ -148,6 +150,15 @@ void HandleExecuteCmdMessage(Message message)
             Reset();
             await Player.Stop();
         }
+
+        void SaveSongToStopAfter(Message songToStopAfter)
```
Comment (theweavrs) [inline BreadPlayer.Views.UWP/ViewModels/ShellViewModel.cs]: Please follow the naming convention for `MessageCallback` i.e. change the method name to `HandleSaveSongToStopAfterMessage`. Thanks.

File: BreadPlayer.Framework/Messenger/MessageTypes.cs [hunk 3]
```diff
@@ -14,7 +14,9 @@ public enum MessageTypes
         MSG_ADD_ALBUMS,
         MSG_PLAYLIST_LOADED,
         MSG_UPDATE_SONG_COUNT,
-        MSG_SEARCH_STARTED
+        MSG_SEARCH_STARTED,
+        MSG_ADD_STOP_SONG
```
Comment (theweavrs) [inline BreadPlayer.Framework/Messenger/MessageTypes.cs]: Make the `MessageType` name a little less inconspicuous. Something like `MSG_STOP_AFTER_SONG` or some such.

**Comment by vsarunov** [comment]
@theweavrs Hi, thanks for the review, will update it as soon as possible :)

**Comment by theweavrs** [comment]
@vsarunov Thanks a lot. Can't wait to merge this :)

**Comment by theweavrs** [comment]
I still don't understand why you are repeating the code to stop the song (line 623 - 627). They can easily be replaced with call to the method `PlayPause()`. Or invoke to command PlayPauseCommand.

**Pull request merged**